sdk/swap-service/src/implementation/oneinch/OneInchSwapProvider.ts

```diff
@@ -0,0 +1,92 @@
+import { Address, Percentage, TokenAmount } from '@summerfi/sdk-common/common'
+import { SwapData, ISwapProvider, SwapProviderType } from '~swap-service/interfaces'
+import { ChainInfo } from '@summerfi/sdk-common/chains'
+import {
+  OneInchAuthHeader,
+  OneInchAuthHeaderKey,
+  OneInchSwapProviderConfig,
+  OneInchSwapResponse,
+} from './types'
+import { Hex } from 'viem'
+import fetch from 'node-fetch'
+
+export class OneInchSwapProvider implements ISwapProvider {
+  public type: SwapProviderType = SwapProviderType.OneInch
+
+  private readonly _apiUrl: string
+  private readonly _version: string
+  private readonly _allowedSwapProtocols: string[]
+  private readonly _apiKey: string
+
+  constructor(params: OneInchSwapProviderConfig) {
+    this._apiUrl = params.apiUrl
+    this._version = params.version
+    this._apiKey = params.apiKey
+    this._allowedSwapProtocols = params.allowedSwapProtocols
+  }
+
+  public async getSwapData(params: {
+    chainInfo: ChainInfo
+    fromAmount: TokenAmount
+    toMinimumAmount: TokenAmount
+    recipient: Address
+    slippage: Percentage
+  }): Promise<SwapData> {
```

Comment:
https://docs.1inch.io/docs/aggregation-protocol/api/swap-params 

One inch api does not support minimum.